geps/gep-3792/index.md

```diff
@@ -0,0 +1,190 @@
+# GEP-3792: External Gateways
+
+* Issue: [#3792](https://github.com/kubernetes-sigs/gateway-api/issues/3792)
+* Status: Provisional
+
+(See [status definitions](../overview.md#gep-states).)
+
+## User Story
+
+**[Chihiro] and [Ian] want a way for out-of-cluster Gateways to be able to
+usefully participate in a GAMMA-compliant in-cluster service mesh.**
+
+Historically, API gateways and ingress controllers have often been implemented
+using a Service of type LoadBalancer fronting a Kubernetes pod running a
+proxy. This is simple to reason about, easy to manage for sidecar meshes, and
+will presumably be an important implementation mechanism for the foreseeable
+future. Some cloud providers, though, are moving the proxy outside of the
+cluster, for various reasons which are out of the scope of this GEP. Chihiro
+and Ian want to be able to use these out-of-cluster proxies effectively and
+safely, though they recognize that this may require additional configuration.
+
+[Chihiro]: https://https//gateway-api.sigs.k8s.io/concepts/roles-and-personas/#chihiro
+[Ian]: https://https//gateway-api.sigs.k8s.io/concepts/roles-and-personas/#ian
+
+### Nomenclature and Background
+
+In this GEP, we will use _out-of-cluster Gateway_ (OCG) to refer to a
+conformant implementation of Gateway API's `GATEWAY` profile that's running
+outside of the cluster. This would most commonly be a managed implementation
+from a cloud provider, but of course there are many other possibilities.
+
+We'll also distinguish between _mTLS meshes_, which rely on standard mTLS for
```

Comment:
I think neither Istio nor Linkerd use standard mTLS really - both have custom protocols (if I read https://github.com/linkerd/linkerd2-proxy/blob/6118756991934f4c33ed51d16c43e1e731164b26/linkerd/transport-header/src/lib.rs#L41 right)

Reply:
Practically, this GEP is going to need to end up for many use cases being "how do we get the Gateway API implementation to behave as an identifiable mesh TLS client without actually injecting config anywhere?". So, because we're configuring how the Gateway will behave as a TLS client, there's some crossover with the BackendTLSPolicy work (to me at least).

Reply:
I added a brief discussion about this into the "protocol problem" section, and added graduation criteria that the problem _must_ be resolved. Obviously we'll hash allllll this out in the API discussion.